sammo.py

```diff
@@ -239,20 +240,67 @@ def onGpsFrame(
         speed: float = -9999.0,
         course: float = -9999.0,
     ) -> None:
-        self.session.lastGpsInfo = (
-            QgsGeometry.fromPointXY(QgsPointXY(longitude, latitude)),
-            (speed, course),
-        )
+        updated = True
         now = datetime.now()
         gpsNow = datetime(now.year, now.month, now.day, h, m, s)
+
+        if self.session.lastGpsInfo[2] == gpsNow and (
+            speed != -9999.0 or course != -9999.0
+        ):
+            # a GPRMC frame is coming after a GPGGA frame with the same
+            # datetime but speed/course are valid
+            self.session.lastGpsInfo[1] = (speed, course)
```

Comment:
SammoSession.lastGPSInfo is a tuple, so we can't re-assigned values. But anyway I wasn't satisfied about it. I'll change it for a dictionnary.

Reply:
done on dev